src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts

```diff
@@ -174,11 +179,34 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 		return content;
 	}
 
+	private getExtensionContributedHomeToolbarContent(content: ITaskbarContent[]): void {
+		let primary: IAction[] = [];
+		let secondary: IAction[] = [];
+		const menu = this.menuService.createMenu(MenuId.DashboardToolbar, this.contextKeyService);
+		let groups = menu.getActions({ arg: null, shouldForwardArgs: true });
+		fillInActions(groups, { primary, secondary }, false, (group: string) => group === undefined || group === '');
```

Comment:
Right now, we were planning on only allowing extensions to contribute actions to the home page toolbar so only actions with no group specified are added to the home page toolbar. However, this could be changed to use the group to specify which dashboard page toolbar an action is added to.

Reply:
This seems like a bit of a limitation if I'm understanding correctly.  If an extension contributes a dashboard page hierarchy the extension can only customize the toolbar on the default page?  What is the toolbar behavior on all the other pages in the contributed page hierarchy?

Reply:
In the latest changes, the context key tabName can now be used to specify which tab the action should show up on. Extensions can customize the toolbar of the default/home tab by using the dashboard/toolbar menu contribution point added in this PR. To customize the toolbar of an extension added tab, the actions can be specified using the tasks-widget or using the dashboard/toolbar menu contribution point.